content/blog/atomic-design-for-spas.md

```diff
@@ -0,0 +1,213 @@
+[app.scss](https://github.com/Keats/hn-react/blob/91bb067276006b0ea9d191f753112a5688b7c518/src/style/app.scss).
+
+If I am using [GraphQL](https://graphql.org/) with [Apollo](https://www.apollographql.com/), I will also have a `queries.tsx` in there with the queries
+used by the component as well as the types:
+
+
+```tsx
+import gql from "graphql-tag";
+import {Query} from "react-apollo";
+
+import {IOrganization} from "../../../types";
+
+export const ALL_ORGS_QUERY = gql`
+  query allOrgs {
+    organizations {
+      id
+      name
+      slug
+
+      members {
+        id
+        role
+
+        user {
+          id
+          name
+          timezone
+        }
+      }
+    }
+  }
+`;
+
+export interface IData {
+  organizations: Array<IOrganization>;
+}
+
+export class AllOrgsQuery extends Query<IData> {}
+```
+
+You can find more about using TypeScript with Apollo [in their docs](https://www.apollographql.com/docs/react/recipes/static-typing.html).
+
+Now that we have seen how I organise each individual component, let's dive in the Atomic Design part.
+
+## Atomic Design
+
+If you have read the article introducing Atomic Design, you can probably guess that your `components` folder is going to look like that:
+
+```bash
+.
+├── atoms
+├── molecules
+├── organisms
+├── templates
+└── pages
+```
+
+Let's look at them in the same order.
+
+### Atoms
+
+Atoms are independent components that do not have any dependencies on other components and that exist in isolation: they
+probably don't do anything on their own, are mostly composed of `props` and any action they do is coming from `props`.
+In terms of Redux/MobX, all of those components should be dumb/pure/presentational components.
+
+Looking at one private project atoms, I have:
+
+- `button`
+- `input`
+- `link`
+- `tag`
+- `toast`
+- `portal`
+- `icon`
+
+There are obviously many more but this should be enough to give you an idea: they are the building blocks
+of your site.
+
+Try to have as many of your components in that folder, you will have a much easier time composing interfaces later on.
+
+### Molecules
+
+By themselves, atoms are not very useful and need to be combined to make something useable: `input` + `button` giving you a `form` for example.
+That is what molecules are: a combination of 2 or more `atoms` or, in rarer cases, a smart wrapper around an atom.
+Therefore molecules should be dumb components when possible but it isn't required.
+
+Here are some of the molecules from the same project:
+
+- `form` (`input` + `label`): using MobX for state but dumb otherwise
+- `dialog` (`overlay` + `button`): dumb
+- `toast-container` (`toast`): smart component that will display the currents `<Toast>` components from the store
+
+Whether `toast-container` belongs in the molecules folder is debatable
+but this is where it makes the most sense in my opinion.
```

Comment:
You could say why. By reading so far it seems like you're doing that because you want all the components in `atoms` to be dumb